
Submitted By: Robert Connolly <robert at linuxfromscratch dot org> (ashes)
Date: 2005-03-20
Initial Package Version: 21.6
Upstream Status: Not submitted - Reported
ORIGIN: Gentoo psmisc-21.6-nonls.patch
Description: uClibc does not have rpmatch(). This patch creates a compatable function.

diff -Naur psmisc-21.6.orig/src/i18n.h psmisc-21.6/src/i18n.h
--- psmisc-21.6.orig/src/i18n.h	2004-11-09 11:25:15.000000000 +0000
+++ psmisc-21.6/src/i18n.h	2005-03-20 09:03:44.000000000 +0000
@@ -15,5 +15,13 @@
 #define _(String) (String)
 #endif
 
+#ifdef __UCLIBC__
+#define rpmatch(line) \
+	( (line == NULL) ? -1 : \
+	  (*line == 'y' || *line == 'Y') ? 1 : \
+	  (*line == 'n' || *line == 'N') ? 0 : \
+	  -1 )
+#endif
+
 #endif
 

